Build failure when using Elemental2 1.2.x

The J2CL maven plugin stops working when using Elemental2 1.2.x. 

### Reproducer

```sh
mvn -B archetype:generate -DarchetypeGroupId=com.vertispan.j2cl.archetypes \
      -DarchetypeArtifactId=j2cl-archetype-simple \
      -DarchetypeVersion=0.21.0 \
      -DgroupId=org.acme \
      -DartifactId=helloworld \
      -Dversion=0.0.1 \
      -Dmodule=helloworld

cd helloworld
mvn package
```

→ No issues

```sh
sed -i 's/<elemental2.version>1.1.0<\/elemental2.version>/<elemental2.version>1.2.1<\/elemental2.version>/g' pom.xml
mvn clean package
```

→ Build error

```
[INFO] Starting com.google.elemental2:elemental2-core:1.2.1/stripped_bytecode_headers
[INFO] com.google.elemental2:elemental2-core:1.2.1/stripped_bytecode_headers: /private/tmp/helloworld/target/gwt3BuildCache/0.21.0/com.google.elemental2-elemental2-core-1.2.1/372a708cb89ddc52b085a53c2185ce46-stripped_sources/results/elemental2/core/BigInt.java:23: error: symbol not found jsinterop.base.JsBigint
import jsinterop.base.JsBigint;
       ^
...

[INFO] Starting com.google.elemental2:elemental2-dom:1.2.1/transpiled_js
[ERROR] com.google.elemental2:elemental2-core:1.2.1/transpiled_js: Error:BigInt.java:23: The import jsinterop.base.JsBigint cannot be resolved
```

--- a/j2cl-maven-plugin/src/it/elemental2-project/pom.xml
+++ b/j2cl-maven-plugin/src/it/elemental2-project/pom.xml
@@ -7,7 +7,7 @@
     <version>1.0</version>
 
     <properties>
-        <elemental2.version>1.1.0</elemental2.version>
+        <elemental2.version>1.2.1</elemental2.version>
     </properties>
 
     <dependencies>

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureBundleTask.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureBundleTask.java
@@ -241,7 +241,7 @@ public class ClosureBundleTask extends TaskFactory {
     public interface SourceSupplier {
         String get() throws IOException;
     }
-    public static class DependencyInfoAndSource extends DependencyInfo.Base {
+    public static class DependencyInfoAndSource implements DependencyInfo {
         private final DependencyInfo delegate;
         private final SourceSupplier sourceSupplier;
 
@@ -291,11 +291,6 @@ public class ClosureBundleTask extends TaskFactory {
         }
 
         @Override
-        public boolean isModule() {
-            return delegate.isModule();
-        }
-
-        @Override
         public boolean isEs6Module() {
             return delegate.isEs6Module();
         }
@@ -316,7 +311,7 @@ public class ClosureBundleTask extends TaskFactory {
         }
     }
 
-    public static class DependencyInfoFormat extends DependencyInfo.Base {
+    public static class DependencyInfoFormat implements DependencyInfo {
         private String name;
 //        private String pathRelativeToClosureBase = name;
         private List<String> provides;

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/GwtIncompatiblePreprocessor.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/GwtIncompatiblePreprocessor.java
@@ -29,7 +29,7 @@ public class GwtIncompatiblePreprocessor {
         Problems problems = new Problems();
 
         try (OutputUtils.Output output = OutputUtils.initOutput(outputDirectory.toPath(), problems)) {
-            GwtIncompatibleStripper.preprocessFiles(unprocessedFiles, output, problems);
+            GwtIncompatibleStripper.preprocessFiles(unprocessedFiles, output, problems, "GwtIncompatible");
 
             if (problems.hasErrors()) {
                 throw new IllegalStateException(problems.getErrors().toString());

--- a/j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/J2cl.java
+++ b/j2cl-tasks/src/main/java/com/vertispan/j2cl/tools/J2cl.java
@@ -46,9 +46,9 @@ public class J2cl {
                     .setOutput(output)
                     .setSources(sourcesToCompile)
                     .setNativeSources(nativeSources)
-                    .setKotlinCommonSources(Collections.emptyList())
                     .setKotlincOptions(ImmutableList.of())
-                    .build();
+                    .setWasmEntryPointStrings(ImmutableList.of())
+                    .build(problems);
 
             log.debug(options.toString());
 

--- a/pom.xml
+++ b/pom.xml
@@ -87,8 +87,8 @@
     <maven.gpg.plugin.version>1.6</maven.gpg.plugin.version>
 
     <!-- Builder dependencies versions -->
-    <j2cl.version>0.11.0-9336533b6</j2cl.version>
-    <closure.compiler.unshaded.version>v20221102-1</closure.compiler.unshaded.version>
+    <j2cl.version>v20230718-1</j2cl.version>
+    <closure.compiler.unshaded.version>v20230411-1</closure.compiler.unshaded.version>
     <commons.codec.version>1.11</commons.codec.version>
     <commons.io.version>2.7</commons.io.version>
 
@@ -115,7 +115,7 @@
     <jetty.version>9.4.43.v20210629</jetty.version>
 
     <!-- Required core j2cl dependencies -->
-    <vertispan.jsinterop.base.version>1.0.0-1</vertispan.jsinterop.base.version>
+    <vertispan.jsinterop.base.version>1.0.1-1</vertispan.jsinterop.base.version>
 
     <!-- Configurations -->
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

